statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,315 @@
+    pca : statsmodels Principal Component Analysis instance
+        The PCA object to use.
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+
+    Returns
+    -------
+    projection : array
+        nobs by nvar array of the projection onto ncomp factors
+
+    """
+    factors = pca.factors
+    pca.factors = data.reshape(-1, factors.shape[1])
+    projection = pca.project()
+    pca.factors = factors
+    return projection
+
+
+def hdrboxplot(data, ncomp=2, alpha=[], threshold=0.95, optimize=False,
+               n_contours=50, xdata=None, labels=None, ax=None):
+    """Plot High Density Region boxplot.
+
+    1. Compute a multivariate kernel density estimation,
+    2. Compute contour lines for percentiles 90%, 50% and `alpha`%,
+    3. Plot the bivariate plot,
+    4. Compute mediane curve along with percentiles and outliers curves.
+
+    Behind the scene, the dataset is represented as a matrix. Each line
+    corresponding to a 1D curve. This matrix is then decomposed using Principal
+    Components Analysis (PCA). This allows to represent the data using a finite
+    number of modes, or components. This compression process allows to turn the
+    functional representation into a scalar representation of the matrix. In
+    other words, you can visualize each curve from its components. With 2
+    components, this is called a bivariate plot.
+
+    This visualization exhibit a cluster of points. It indicate that a lot of
+    curve lead to common components. The center of the cluster is the mediane
+    curve. An the more you get away from the cluster, the more the curve is
+    unlikely to be similar to the other curves.
+
+    Using a kernel smoothing technique, the probability density function (PDF)
+    of the multivariate space can be recover. From this PDF, it is possible to
+    compute the density probability linked to the cluster and plot its
+    contours.
+
+    Finally, using these contours, the different quantiles can be extracted
+    allong with the median curve and the outliers.
+
+    Parameters
+    ----------
+    data : sequence of ndarrays or 2-D ndarray
+        The vectors of functions to create a functional boxplot from.  If a
+        sequence of 1-D arrays, these should all be the same size.
+        The first axis is the function index, the second axis the one along
+        which the function is defined.  So ``data[0, :]`` is the first
+        functional curve.
+    ncomp : int, optional
+        Number of components to use.  If None, returns the as many as the
+        smaller of the number of rows or columns in data.
+    alpha : list, optional
+        Extra percentile values to compute.
```

Comment:
use `quantile` instead of percentile, to emphasize that we want a fractional number between 0 and 1.

same for threshold
"fraction" also works